deploy/deploy.sh

```diff
@@ -16,54 +16,122 @@
+]
+EOS
+  cat <<EOS >> ${KUSTOMIZATION_FILE}
+patchesJson6902:
+  - path: ${FILE}
+    target:
+      group: apps
+      version: v1
+      kind: DaemonSet
+      namespace: openshift-cluster-csi-drivers
+      name: csi-hostpathplugin
+EOS
+}
+
+# uses `oc wait` to wait for CSI driver pod to reach condition ready.
+function wait_for_pod () {
+  oc --namespace="openshift-cluster-csi-drivers" wait pod \
+    --for="condition=Ready=true" \
+    --selector="app=csi-hostpathplugin" \
+    --timeout="5m"
+}
+
+echo "# Creating deploy directory at '${DEPLOY_DIR}'"
+
+rm -rf "${DEPLOY_DIR}" || true
 mkdir -p "${DEPLOY_DIR}"
 
-defaultRegistrarImage="quay.io/openshift/origin-csi-node-driver-registrar:4.9.0"
-registrarImage=${NODE_REGISTRAR_IMAGE:-${defaultRegistrarImage}}
-defaultDriverImage="quay.io/openshift/origin-csi-driver-shared-resource:4.9.0"
-driverImage=${DRIVER_IMAGE:-${defaultDriverImage}}
+cp -r -v "${BASE_DIR}"/* "${DEPLOY_DIR}"
+
+echo "# Customizing resources..."
+
+# initializing kustomize and adding the all resource files it should use
+kustomize_init
 
-cp -r "${BASE_DIR}"/* "${DEPLOY_DIR}"
+if [ ! -z "${NODE_REGISTRAR_IMAGE}" ] ; then
+  echo "# Patching node-registrar image to use '${NODE_REGISTRAR_IMAGE}'"
+  kustomize_set_image "quay.io/openshift/origin-csi-node-driver-registrar" "${NODE_REGISTRAR_IMAGE}"
+fi
 
-echo "Deploying using node driver registrar ${registrarImage}"
-echo "Deploying using csi driver ${driverImage}"
+if [ ! -z "${DRIVER_IMAGE}" ] ; then
+  echo "# Patching node-csi-driver image to use '${DRIVER_IMAGE}'"
+  kustomize_set_image "quay.io/openshift/origin-csi-driver-shared-resource" "${DRIVER_IMAGE}"
+fi
 
-# Replace the image refs in the canonical deployment YAML with env var overrides
-sed -i -e "s|${defaultRegistrarImage}|${registrarImage}|g" \
-  -e "s|${defaultDriverImage}|${driverImage}|g" \
-  "${DEPLOY_DIR}/csi-hostpath-plugin.yaml"
+# adding the read-only flag by using kustomize-v3 approach (embedded on `oc`)
+if [ ! -z "${READ_ONLY_MODE}" ] ; then
+  echo "# Patching DaemonSet container to use --readonly flag"
+  kustomize_add_arg "--readonly"
+fi
 
 # deploy hostpath plugin and registrar sidecar
-echo "deploying csi driver components"
-for i in $(find "${DEPLOY_DIR}"/*.yaml | sort); do
-    echo "   $i"
-    run oc apply -f "$i"
-done
-
-# Wait until all pods are running.
-expected_running_pods=3
-cnt=0
-while [ "$(oc get pods -n openshift-cluster-csi-drivers 2>/dev/null | grep -c '^csi-hostpath.* Running ')" -lt ${expected_running_pods} ]; do
-    if [ $cnt -gt 30 ]; then
-        echo "$(oc get pods 2>/dev/null | grep -c '^csi-hostpath.* Running ') running pods:"
-        oc describe pods
-
-        echo >&2 "ERROR: hostpath deployment not ready after over 5min"
-        exit 1
-    fi
-    echo "$(date +%H:%M:%S) waiting for hostpath deployment to complete, attempt #$cnt"
-    cnt=$((cnt + 1))
-    sleep 10
-done
+echo "# Deploying csi driver components"
+run oc apply --kustomize ${DEPLOY_DIR}/
```

Comment:
ok @otaviof is able to use `kustomize` via `oc apply` @adambkaplan 

so I think this closes the loop on this discussion thread

we'll use this vs. `sed`